### PR TITLE
SOLR-17489: CLI: Deprecate variations on solr urls options

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-17489

# Description

See JIRA

# Solution

Deprecate old options in favour of `--solr-url` and `--name`.   I am really liking how this looks after going through the code...

Debating if we should be using `--collection` instead of `--name`....   That can be a seperate ticket.

# Tests

bats and java

### CODE REVIEW

**Comment by epugh** [comment]
@malliaridis  okay, this one is ready for another set of eyes!

**Comment by epugh** [comment]
After merging this, we can then sort out the `-s` in the SolrExporter.java class.

File: solr/core/src/test/org/apache/solr/cli/PostToolTest.java
Comment (malliaridis) [inline solr/core/src/test/org/apache/solr/cli/PostToolTest.java:89]: `getBaseUrl()` extends v1 API path `/solr`, and the new logic does also add `/solr/` if `--solr-url` is used. This results to a path like `/solr/solr/collection/update`.

File: solr/core/src/test/org/apache/solr/cli/TestExportTool.java
Comment (malliaridis) [inline solr/core/src/test/org/apache/solr/cli/TestExportTool.java:247]: Same issue with path here.

File: solr/core/src/test/org/apache/solr/cloud/SolrCloudExampleTest.java
Comment (malliaridis) [inline solr/core/src/test/org/apache/solr/cloud/SolrCloudExampleTest.java:119]: Same problem with path applies here as well, but this uses different logic.

**Review by malliaridis — CHANGES_REQUESTED** [review]
I like the fact that we can use the same params in all tools now. I hope this change is preferred by the others as well. A few more references:

- [ ] The error message in PostTool#runImpl (line 309) could be updated to reflect the none-deprecated options
- [x] post-tool.adoc#27: -url should be split into --solr-url and --name
- [x] post-tool.adoc#115: -url should be split into --solr-url and --name
- [x] post-tool.adoc#124: -url should be split into --solr-url and --name
- [ ] post-tool.adoc#131: -url should be split into --solr-url and --name
- [x] post-tool.adoc#138: -url should be split into --solr-url and --name
- [x] post-tool.adoc#154: -url should be split into --solr-url and --name
- [x] post-tool.adoc#166: -url should be split into --solr-url and --name
- [x] post-tool.adoc#173: -url should be split into --solr-url and --name
- [x] post-tool.adoc#180: -url should be split into --solr-url and --name
- [x] post-tool.adoc#189: -url should be split into --solr-url and --name
- [x] post-tool.adoc#210: -url should be split into --solr-url and --name
- [x] post-tool.adoc#219: -url should be split into --solr-url and --name
- [x] spatal-search.adoc#74: -url should be split into --solr-url and --name

File: solr/core/src/test/org/apache/solr/cli/PostToolTest.java
Comment (epugh) [reply solr/core/src/test/org/apache/solr/cli/PostToolTest.java:89]: i think I fixed it with the use of `SolrCLI.normalizeSolrUrl() in the tool`

**Comment by epugh** [comment]
Will add CHANGES.txt all at once.

**Pull request merged**
